source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1142,8 +1166,12 @@ void RtFileIo::read(const FilePath& documentPath, const FileSearchPath& searchPa
             xmlReadOptions.desiredMinorVersion = readOptions->desiredMinorVersion;
         }
         readFromXmlFile(document, documentPath, searchPaths, &xmlReadOptions);
+
         string errorMessage;
-        if (document->validate(&errorMessage))
+        DocumentPtr validationDocument = createDocument();
+        writeUnitDefinitions(validationDocument);
```

Comment:
Units added to a "validation" document. This avoids embedding units into the document.